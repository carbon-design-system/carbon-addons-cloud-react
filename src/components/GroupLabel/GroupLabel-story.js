import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import GroupLabel from './GroupLabel';
import { Tooltip } from 'carbon-components-react';

const additionalProps = {
  className: 'some-class',
};

storiesOf('GroupLabel', module)
  .add(
    'Default',
    withInfo({
      text: 'Group label.',
    })(() => <GroupLabel {...additionalProps}>Label</GroupLabel>)
  )
  .add(
    'With tooltip',
    withInfo({
      text: 'Form label with tooltip.',
    })(() => (
      <GroupLabel {...additionalProps}>
        <Tooltip triggerText="Label">
          This is the content of the tooltip.
        </Tooltip>
      </GroupLabel>
    ))
  );
