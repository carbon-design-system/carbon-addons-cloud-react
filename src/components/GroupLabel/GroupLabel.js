import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';

/*const categoryLabel = {
  color: '#8897A2',
  fontSize: '12px',
  fontFamily: 'ibm plex Sans',
  fontWeight: '700',
  letterSpacing: '0.2px',
  margin: '8px',
};*/

const GroupLabel = ({ className, children, id, ...other }) => {
  const classNames = classnames('bx--group-label', className);

  return (
    <label htmlFor={id} className="bx--group-label" {...other}>
      {children}
    </label>
  );
};

GroupLabel.propTypes = {
  /**
   * Specify the content of the form label
   */
  children: PropTypes.node,

  /**
   * Provide a custom className to be applied to the containing <label> node
   */
  className: PropTypes.string,

  /**
   * Provide a unique id for the given <GroupLabel>
   */
  id: PropTypes.string,
};

export default GroupLabel;
