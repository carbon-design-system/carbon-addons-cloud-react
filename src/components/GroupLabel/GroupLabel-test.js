import React from 'react';
import { shallow } from 'enzyme';
import GroupLabel from '../GroupLabel';

describe('GroupLabel', () => {
  it('should render', () => {
    const wrapper = shallow(<GroupLabel />);
    expect(wrapper).toMatchSnapshot();
  });
});
