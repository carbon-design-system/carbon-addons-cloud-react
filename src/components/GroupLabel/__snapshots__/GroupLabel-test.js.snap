// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupLabel should render 1`] = `
<label
  className="bx--group-label"
/>
`;
